src/app/beer_garden/db/mongo/util.py

```diff
@@ -93,21 +94,72 @@ def _configure_superuser_role():
     superuser.save()
 
 
+def _configure_plugin_role():
+    """Creates or updates the plugin role as needed"""
+    try:
+        plugin_role = Role.objects.get(name="plugin")
+    except Role.DoesNotExist:
+        logger.info("Creating plugin role with select permissions")
+        plugin_role = Role(name="plugin")
+
+    plugin_role.permissions = [
+        Permissions.GARDEN_READ.value,
+        Permissions.INSTANCE_READ.value,
+        Permissions.INSTANCE_UPDATE.value,
+        Permissions.REQUEST_CREATE.value,
+        Permissions.REQUEST_READ.value,
+        Permissions.REQUEST_UPDATE.value,
+        Permissions.SYSTEM_READ.value,
+        Permissions.SYSTEM_UPDATE.value,
+    ]
+    plugin_role.description = "Role containing plugin permissions"
+    plugin_role.protected = True
+
+    plugin_role.save()
+
+
 def ensure_users():
-    """Create the default admin user if necessary"""
+    """Create user accounts if necessary"""
     if User.objects.count() == 0:
-        username = config.get("auth.default_admin.username")
-        password = config.get("auth.default_admin.password")
-        superuser_role = Role.objects.get(name="superuser")
+        _create_admin()
+        _create_plugin_user()
+
+
+def _create_admin():
+    """Create the default admin user if necessary"""
+    username = config.get("auth.default_admin.username")
+    password = config.get("auth.default_admin.password")
+    superuser_role = Role.objects.get(name="superuser")
+
+    logger.info("Creating default admin user with username: %s", username)
+
+    admin = User(username=username)
+    admin.set_password(password)
+    admin.role_assignments = [
+        RoleAssignment(role=superuser_role, domain={"scope": "Global"})
+    ]
+    admin.save()
+
+
+def _create_plugin_user():
+    """Create the default user to run Plugins if necessary"""
+    username = config.get("plugin.local.auth.username")
+    plugin_user = User.objects(username=username).first()
+
+    # Sanity check to make sure we don't accidentally create two
+    # users with the same name
+    if not plugin_user:
+        password = config.get("plugin.local.auth.password")
+        plugin_user_role = Role.objects.get(name="plugin")
 
-        logger.info("Creating default admin user with username: %s", username)
+        logger.info("Creating default plugin user with username: %s", username)
 
-        admin = User(username=username)
-        admin.set_password(password)
-        admin.role_assignments = [
-            RoleAssignment(role=superuser_role, domain={"scope": "Global"})
+        plugin_user = User(username=username)
+        plugin_user.set_password(password)
+        plugin_user.role_assignments = [
+            RoleAssignment(role=plugin_user_role, domain={"scope": "Local"})
```

Comment:
The scope here should be "Global".  The valid scopes are:

* Global
* Garden
* System

Some info about that is documented [here](https://beer-garden.io/docs/app/security/#role_assignments)